### PR TITLE
bug_fix

### PR DESCRIPTION
**Pull request by kRhythm53**

### CODE REVIEW

File: types/downtimes/database.go [hunk 1]
```diff
@@ -67,6 +67,13 @@ func FindDowntime(timeVar time.Time) []Downtime {
 	return downtime
 }
 
+func FindLatestDowntimeOfService(service int64) Downtime {
+	var downtime Downtime
+	q := db.Where("service = ?", service)
+	q = q.Order("start desc").First(&downtime)
```
Comment (dmonark) [inline types/downtimes/database.go:73]: End Desc

**Review by dmonark — CHANGES_REQUESTED** [review]
Run go fmt please

File: types/downtimes/database.go (hunk 1 above)
Comment (kRhythm53) [reply types/downtimes/database.go:73]: okay will change it.

File: types/downtimes/database.go [hunk 2]
```diff
@@ -67,6 +67,13 @@ func FindDowntime(timeVar time.Time) []Downtime {
 	return downtime
 }
 
+func FindLatestDowntimeOfService(service int64) Downtime {
+	var downtime Downtime
+	q := db.Where("service = ? ", service)
+	q = q.Order("start desc ").First(&downtime)
```
Comment (dmonark) [inline types/downtimes/database.go]:
```suggestion
	q = q.Order("start desc").First(&downtime)
```

File: types/downtimes/database.go [hunk 3]
```diff
@@ -53,7 +53,7 @@ func (c *Downtime) BeforeUpdate() error {
 	return c.Validate()
 }
 
-func FindByService(service int64, start time.Time, end time.Time) (*[]Downtime, error) {
+func FindByServiceAndDuration(service int64, start time.Time, end time.Time) (*[]Downtime, error) {
 	var downtime []Downtime
 	q := db.Where("service = ? and start BETWEEN ? AND ? ", service, start, end)
 	q = q.Order("id ASC ").Find(&downtime)
```
Comment (dmonark) [inline types/downtimes/database.go]:
```suggestion
	q = q.Order("id ASC").Find(&downtime)
```

File: types/downtimes/database.go [hunk 4]
```diff
@@ -53,7 +53,7 @@ func (c *Downtime) BeforeUpdate() error {
 	return c.Validate()
 }
 
-func FindByService(service int64, start time.Time, end time.Time) (*[]Downtime, error) {
+func FindByServiceAndDuration(service int64, start time.Time, end time.Time) (*[]Downtime, error) {
 	var downtime []Downtime
 	q := db.Where("service = ? and start BETWEEN ? AND ? ", service, start, end)
```
Comment (dmonark) [inline types/downtimes/database.go]:
```suggestion
	q := db.Where("service = ? and start BETWEEN ? AND ?", service, start, end)
```

File: types/downtimes/database.go [hunk 5]
```diff
@@ -67,6 +67,13 @@ func FindDowntime(timeVar time.Time) []Downtime {
 	return downtime
 }
 
+func FindLatestDowntimeOfService(service int64) Downtime {
+	var downtime Downtime
+	q := db.Where("service = ? ", service)
```
Comment (dmonark) [inline types/downtimes/database.go]:
```suggestion
	q := db.Where("service = ?", service)
```

**Review by dmonark — CHANGES_REQUESTED** [review]

**Review by dmonark — APPROVED** [review]

**Pull request merged**